internal/impl/postgresql/input_postgrecdc.go

```diff
@@ -0,0 +1,562 @@
+
+type asyncMessage struct {
+	msg   service.MessageBatch
+	ackFn service.AckFunc
+}
+
+var pgStreamConfigSpec = service.NewConfigSpec().
+	Beta().
+	Categories("Services").
+	Version("0.0.1").
+	Summary(`Creates a PostgreSQL replication slot for Change Data Capture (CDC)
+		== Metadata
+
+This input adds the following metadata fields to each message:
+- streaming (Indicates whether the message is part of a streaming operation or snapshot processing)
+- table (Name of the table that the message originated from)
+- operation (Type of operation that generated the message, such as INSERT, UPDATE, or DELETE)
+		`).
+	Field(service.NewStringField("host").
+		Description("The hostname or IP address of the PostgreSQL instance.").
+		Example("123.0.0.1")).
+	Field(service.NewIntField("port").
+		Description("The port number on which the PostgreSQL instance is listening.").
+		Example(5432).
+		Default(5432)).
+	Field(service.NewStringField("user").
+		Description("Username of a user with replication permissions. For AWS RDS, this typically requires superuser privileges.").
+		Example("postgres"),
+	).
+	Field(service.NewStringField("password").
+		Description("Password for the specified PostgreSQL user.")).
+	Field(service.NewStringField("schema").
+		Description("The PostgreSQL schema from which to replicate data.")).
+	Field(service.NewStringField("database").
+		Description("The name of the PostgreSQL database to connect to.")).
+	Field(service.NewStringEnumField("tls", "require", "none").
+		Description("Specifies whether to use TLS for the database connection. Set to 'require' to enforce TLS, or 'none' to disable it.").
+		Example("none").
+		Default("none")).
+	Field(service.NewBoolField("stream_uncomited").
+		Description("If set to true, the plugin will stream uncommitted transactions before receiving a commit message from PostgreSQL. This may result in duplicate records if the connector is restarted.").
+		Default(false)).
+	Field(service.NewStringField("pg_conn_options").
+		Description("Additional PostgreSQL connection options as a string. Refer to PostgreSQL documentation for available options.").
+		Default(""),
+	).
+	Field(service.NewBoolField("stream_snapshot").
+		Description("When set to true, the plugin will first stream a snapshot of all existing data in the database before streaming changes.").
+		Example(true).
+		Default(false)).
+	Field(service.NewFloatField("snapshot_memory_safety_factor").
+		Description("Determines the fraction of available memory that can be used for streaming the snapshot. Values between 0 and 1 represent the percentage of memory to use. Lower values make initial streaming slower but help prevent out-of-memory errors.").
+		Example(0.2).
+		Default(1)).
+	Field(service.NewIntField("snapshot_batch_size").
+		Description("The number of rows to fetch in each batch when querying the snapshot. A value of 0 lets the plugin determine the batch size based on `snapshot_memory_safety_factor` property.").
```

Comment:
We should instead of this make them unset so we can easily check for mutually exclusive as a lint check:

https://github.com/redpanda-data/connect/blob/754fd8a7e54edc44e22affca29af0fee41b52235/internal/impl/snowflake/output_snowflake_streaming.go#L98-L100

You can check if it's set using `.Contains` and default to the appropriate value if neither is set when extracting the config.